src/tools/clippy/clippy_lints/src/ranges.rs

```diff
@@ -356,6 +355,8 @@ fn check_exclusive_range_plus_one(cx: &LateContext<'_>, expr: &Expr<'_>) {
             end: Some(end),
             limits: RangeLimits::HalfOpen
         }) = higher::Range::hir(expr);
+        if start.map_or(true, |start| start.span.can_be_used_for_suggestions());
+        if end.span.can_be_used_for_suggestions();
```

Comment:
This suppresses the entire lint, if the suggestion can't be build from the spans. Would it be better to just not suggest anything, but still trigger the lint?